From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Quentin Perret <qperret@google.com>
Date: Mon, 30 Sep 2019 16:23:52 +0100
Subject: ANDROID: allmodconfig: Force gki_defconfig as base

Allmodconfig enables as many options as it can to maximize the number of
modules it can build. While this is generally a good idea, this can also
lead to very convoluted configurations (such as CPU_BIG_ENDIAN=y on
arm64) which we don't necessarily want to support even if they break. On
the other hand, gki_defconfig already contains the set of core options
we definitely want to support, so it makes sense to use that as a base.

Point KCONFIG_ALLMODCONFIG to the relevant gki_defconfig file in order
to ensure a minimally sensible config that allconfig is not allowed to
modify.

Bug: 140224784
Signed-off-by: Quentin Perret <qperret@google.com>
Change-Id: Ib4cf3c9565f040a577ce3cec008293520be1af84
---
 build.config.allmodconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build.config.allmodconfig b/build.config.allmodconfig
index 43b1a70d5800..6e0404b8de47 100644
--- a/build.config.allmodconfig
+++ b/build.config.allmodconfig
@@ -1,4 +1,5 @@
 DEFCONFIG=allmodconfig
+KCONFIG_ALLCONFIG=${ROOT_DIR}/common/arch/${ARCH%_*}/configs/gki_defconfig
 
 # XFS_FS is currently broken on this branch with clang-9
 POST_DEFCONFIG_CMDS="update_config"
